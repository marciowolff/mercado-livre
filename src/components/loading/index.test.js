import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Loading from './index'

it('Loading should be a function', () => {
  expect(typeof Loading).toBe('function')
})

it('Should List to match snapshot', () => {
  const tree = renderer.create(<Loading show={true}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should not render Loading', () => {
  expect(mount(<Loading />).find('.loading')).toHaveLength(0)
})

it('Should render Loading', () => {
  const wrapper = mount(<Loading show={true} />)

  expect(wrapper.find('span.loading').hasClass('loading--icon')).toBeTruthy()
  expect(wrapper.find('svg.ico')).toHaveLength(1)
  expect(wrapper.find('small').text()).toEqual('carregando...')
})

it('Should render Loading add className', () => {
  const wrapper = mount(<Loading show={true} className="teste" />)

  expect(wrapper.find('span.loading').hasClass('teste')).toBeTruthy()
})

it('Should render Load without message', () => {
  const wrapper = mount(<Loading show={true} hideMessage={true} />)

  expect(wrapper.find('small')).toHaveLength(0)
})
