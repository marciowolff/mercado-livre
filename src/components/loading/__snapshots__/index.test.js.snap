// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should List to match snapshot 1`] = `
<span
  className="loading  loading--icon"
  id=""
>
  <svg
    className="ico loading "
    fill="none"
    height="80"
    stroke="currentColor"
    viewBox="-4 -4 46 46"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        strokeWidth="5"
        transform="translate(1 1)"
      >
        <circle
          cx="18"
          cy="18"
          r="18"
          strokeOpacity=".5"
        />
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
        >
          <animateTransform
            attributeName="transform"
            dur="1s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
      </g>
    </g>
  </svg>
  <small>
    carregando...
  </small>
</span>
`;
